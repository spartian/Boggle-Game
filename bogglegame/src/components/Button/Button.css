/* Button States */
.button {
    width: 100%;
    border-radius: 5px;
    background-color: #FF9800;
    margin-top: 25px;
    padding: 20px 0;
    -webkit-transition: background 0.5s; /* Safari */
    transition: background 0.5s;
    box-shadow: 2px 2px 5px rgba(255, 204, 128, .25);
    display: block;
    border: none;
    font-size: 15px;
    font-weight: 600;
    text-align: center;
    color: #ffffff;
    letter-spacing: 3px;
}

.button:hover {
    background: #EF6C00;
    cursor: pointer;
}
