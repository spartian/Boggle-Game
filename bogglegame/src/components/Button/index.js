import React from "react";
import "./Button.css";

const Button = props => {
  // TODO: Destructure Button props
  const { handleSubmit, label } = props;

  return (
    <button
      className="button"
      // TODO: Handle on click callback
      onClick={handleSubmit}
    >
      {/* TODO: Render button label */}
      {label}
    </button>
  );
};

export default Button;
