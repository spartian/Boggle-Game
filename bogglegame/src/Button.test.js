import React from 'react';
import expect from 'expect';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../src/components/Button';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

let [onClick] = new Array(1).fill(jest.fn());
const enzymeWrapper = shallow(<Button handleSubmit={onClick}  class="button" label="Hello"></Button>);

describe('Snapshot test of button', () => {

    it('should render a button', () => {
        const component = renderer.create(<Button class="button" />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});


describe('Button assertion test activites', () => {

    it('should render a button with a class', () => {

        let button = enzymeWrapper.find('.button');
        button.simulate('click');

        // Check if the button is clicked once
        expect(onClick.mock.calls.length).toEqual(1);

        // Check if the button has class .button
        expect(button.hasClass('button')).toBe(true);
        
        // Check if the button has label hello
        expect(button.text()).toEqual('Hello');

    });
});