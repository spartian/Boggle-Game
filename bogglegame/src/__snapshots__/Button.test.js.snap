// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of button should render a button 1`] = `
<button
  className="button"
/>
`;
